test/indexer/test_hca_indexer.py

```diff
@@ -1335,6 +1335,87 @@ def test_multiple_samples(self):
                 for sample in contents['samples']:
                     self.assertIn(sample['entity_type'], sample_entity_types)
 
+    def test_sample_with_no_donor(self):
+        """
+        Index two bundles for the same project, one bundle has a sample that
+        is connected to a donor and the other bundle has a sample that is not.
+        Verify the lack of a donor is represented in the project aggregate.
+        """
+        # Sample (Specimen): 70d2b85a, Donor: b111e5bf
+        bundle_fqid = self.bundle_fqid(uuid='1fd499c5-f397-4bff-9af0-eb42c37d5fbe',
+                                       version='2021-03-18T11:38:49.884Z')
+        self._index_canned_bundle(bundle_fqid)
+        # Sample (Organoid): df23c109, Donor: none
+        bundle_fqid = self.bundle_fqid(uuid='0722b70c-6778-423d-8fe9-869e2a515d35',
+                                       version='2021-03-18T11:38:49.863Z')
+        self._index_canned_bundle(bundle_fqid)
+        donor = [
+            {
+                'document_id': 'b111e5bf-e907-47f9-8eed-75b2ec5536c5',
+                'biomaterial_id': 'Human_62',
+                'biological_sex': 'male',
+                'genus_species': ['Homo sapiens'],
+                'development_stage': 'human adult stage',
+                'diseases': ['normal'],
+                'organism_age': '62 year',
+                'organism_age_value': '62',
+                'organism_age_unit': 'year',
+                'organism_age_range': {
+                    'gte': 1955232000.0,
+                    'lte': 1955232000.0
+                }
+            }
+        ]
+        # FIXME: The donor inner entity for a project without a donor
+        #        should have all the standard fields of a donor inner
+        #        entity with values of `None`.
+        #        https://github.com/databiosphere/azul/issues/3152
+        if False:
+            donor_none = [
+                {
+                    key: [None] if isinstance(val, list) else None
+                    for key, val in one(donor).items()
+                    if key != 'organism_age_range'
+                }
+            ]
+        else:
+            donor_none = []
+        # FIXME: The donor inner entity of a project aggregate that
+        #        includes a project without a donor should include
+        #        `None` in each of the aggregated donor fields.
+        #        https://github.com/databiosphere/azul/issues/3152
+        if False:
+            aggregate_donor = {
+                key: val + [None] if isinstance(val, list) else [val, None]
+                for key, val in one(donor).items()
+            }
+            # The `organism_age_range` field will not have a `None`
+            # value since the field is only added to the donor
+            # inner entity if it has `organism_age_in_seconds` value.
+            aggregate_donor['organism_age_range'].pop()
```

Comment:
Brittle and obscure. It is not guaranteed that this pops the `None` entry and not immediately apparent.